src/api-binder.ts

```diff
@@ -450,14 +450,24 @@ export class APIBinder {
 
 	private async sendReportPatch(
 		stateDiff: DeviceApplicationState,
-		conf: { apiEndpoint: string; uuid: string },
+		conf: { apiEndpoint: string; uuid: string; localMode: boolean },
 	) {
 		if (this.cachedBalenaApi == null) {
 			throw new InternalInconsistencyError(
 				'Attempt to send report patch without an API client',
 			);
 		}
 
+		let body = stateDiff;
+		if (conf.localMode) {
+			body = this.stripDeviceStateInLocalMode(stateDiff);
```

Comment:
I was also exploring a possibility to do this `strip` on a collected state rather than on the diff, so that we can make diff on potentially smaller data object. But since there is a lot of logic that both `dependent` and `local` fields have to be present on state objects, everything becomes slightly more complicated. So, I ended up with this approach.